### PR TITLE
Clear metrics when offramp session ends

### PR DESCRIPTION
**Pull request by Noddy76**
Also defensively check gauges for latency. We noticed that if these came from a previous session then because the source object they get the value from is only held by a weak reference then no only will the values we're adding not be the ones that get presented via micrometer but after the object is garbage collected micrometer will report NaN as the value.

### CODE REVIEW

File: endpoint/offramp/service/src/main/java/com/hotels/road/offramp/metrics/OfframpMetrics.java [hunk 1]
```diff
@@ -105,11 +107,32 @@ public void markMessageLatency(int partition, long timestamp) {
     getPartitionLatencyHolder(partition).set(clock.millis() - timestamp);
   }
 
+  @Override
+  public void close() throws Exception {
+    registry.remove(messageCounter);
```
Comment (nahguam) [inline endpoint/offramp/service/src/main/java/com/hotels/road/offramp/metrics/OfframpMetrics.java]: Most of these metrics are granular to the stream level (ie not down to partition level). So if there are multiple connections for the same stream then closing one will remove the metrics for the other. 

Introduce partition tag? Don't remove these ones and tackle in a separate PR?

File: pom.xml [hunk 2]
```diff
@@ -59,7 +59,7 @@
     <reactor-addons.version>3.1.3.RELEASE</reactor-addons.version>
     <ryantenney-metrics.version>3.1.3</ryantenney-metrics.version>
     <slf4j.version>1.7.25</slf4j.version>
-    <spring-boot.version>2.0.1.RELEASE</spring-boot.version>
+    <spring-boot.version>2.1.1.RELEASE</spring-boot.version>
```
Comment (nahguam) [inline pom.xml]: not `2.1.2.RELEASE`?

File: endpoint/offramp/service/src/main/java/com/hotels/road/offramp/metrics/OfframpMetrics.java (hunk 1 above)
Comment (Noddy76) [reply endpoint/offramp/service/src/main/java/com/hotels/road/offramp/metrics/OfframpMetrics.java]: Resolved by using the metric pool

File: pom.xml (hunk 2 above)
Comment (Noddy76) [reply pom.xml]: merged master to resolve

**Pull request merged**